packages/run-protocol/src/vpool-xyk-amm/swap.js

```diff
@@ -19,23 +25,28 @@ export const makeMakeSwapInvitation = (zcf, provideVPool) => {
       give: { In: amountIn },
       want: { Out: amountOut },
     } = seat.getProposal();
-    const pool = provideVPool(amountIn.brand, amountOut.brand).externalFacet;
-    return pool.swapIn(seat, amountIn, amountOut);
+    const pool = provideVPool(amountIn.brand, amountOut.brand).internalFacet;
+    let prices;
+    const maxOut = args.maxOut;
+    if (maxOut) {
+      AmountMath.coerce(amountOut.brand, maxOut);
+      prices = pool.getPriceForOutput(amountIn, maxOut);
+    }
+    if (!prices || !AmountMath.isGTE(prices.swapperGets, maxOut)) {
```

Comment:
I think there are scenarios where current calls to `swapOut()` would fail to trade, and this code would fall through to `getPriceForInput()` and sell anyway. One I found is if there is insufficient liquidity to satisfy the `swapOut`. In that case, I think the failure to trade is the correct outcome, even though that's not what you want in the liquidation case.

I would also like to see reasonably thorough testing of the `swapIn(amountIn, { maxOut })` cases.

Reply:
Since it also passes the `want.Out` as a `want.Out` I believe in _all_ those cases it should properly fail to reallocate because the want is not satisfied.